src/transformers/adapters/peft_mixin.py

```diff
@@ -0,0 +1,333 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import inspect
+from typing import Optional
+
+from ..utils import (
+    check_peft_version,
+    find_adapter_config_file,
+    is_accelerate_available,
+    is_peft_available,
+    logging,
+)
+
+
+if is_accelerate_available():
+    from accelerate import dispatch_model
+    from accelerate.utils import get_balanced_memory, infer_auto_device_map
+
+
+logger = logging.get_logger(__name__)
+
+
+class PeftAdapterMixin:
+    """
+    A class containing all functions for loading and using adapters weights that are supported in PEFT library.
+    Currently supported PEFT methods are all non-prefix tuning methods.
+
+    With this mixin, if the correct PEFT version is installed, it is possible to:
+
+    - Load an adapter stored on a local path or in a remote Hub repository, and inject it in the model
+    - Attach new adapters in the model and train them with Trainer or by your own.
+    - Attach multiple adapters and iteratively activate / deactivate them
+    - Activate / deactivate all adapters from the model.
+    - Get the `state_dict` of the active adapter.
+    """
+
+    _hf_peft_config_loaded = False
+
+    def load_adapter(
+        self,
+        peft_model_id: str,
+        adapter_name: Optional[str] = None,
+        revision: Optional[str] = None,
+        use_auth_token: Optional[str] = None,
+        commit_hash: Optional[str] = None,
+        device_map: Optional[str] = "auto",
+        max_memory: Optional[str] = None,
+        offload_dir: Optional[str] = None,
+        offload_index: Optional[int] = None,
+    ) -> None:
+        """
+        Load adapter weights from file or remote Hub folder. Requires peft as a backend to load the adapter weights.
+
+        Args:
+            peft_model_id (`str`):
+                The identifier of the model to look for on the Hub, or a local path to the saved adapter config file
+                and adapter weights.
+            adapter_name (`str`, `optional`):
+                The adapter name to use. If not set, will use the default adapter.
+            revision (`str`, `optional`):
+                revision argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            use_auth_token (`str`, `optional`):
+                use_auth_token argument to be passed to `hf_hub_download` method from `huggingface_hub`.
+            commit_hash (`str`, `optional`):
+                commit_hash argument to be passed to `hf_hub_download` method from `huggingface_hub`.
```

Comment:
Here you should copy the docstrings from `from_pretrained` to be consistent.

Reply:
Done!